sagan-common/src/main/java/sagan/projects/support/Version.java

```diff
@@ -0,0 +1,105 @@
+package sagan.projects.support;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class Version {
+    private final String source;
+    private final String[] parts;
+    private final Version baseVersion;
+
+    private Version(String source, List<String> parts, Version baseVersion) {
+        this.source = source;
+        this.parts = parts.toArray(new String[0]);
+        this.baseVersion = baseVersion == null ? this : baseVersion;
+    }
+
+    /**
+     * Adapted from Gradle's VersionParser
+     */
+    public static Version build(String raw) {
```

Comment:
I'm not going to pretend I can read this, because it's quite intense. (I'm assuming there are plenty of test cases to insure things.) A handful of comments that would help navigate it would be nice.